drivers/sdcard_spi/include/sdcard_spi_internal.h

```diff
@@ -0,0 +1,241 @@
+/*
+ * Copyright (C) 2016 Michel Rottleuthner <michel.rottleuthner@haw-hamburg.de>
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     drivers_sdcard_spi
+ * @{
+ *
+ * @file
+ * @brief       Interface for issuing commands on sd-cards via SPI.
+ * @detail      For details of the sd card standard and the spi mode refer to
+ *              "SD Specifications Part 1 Physical Layer Simplified Specification".
+ *              References to the sd specs in this file apply to Version 5.00
+ *              from August 10, 2016. For further details see
+ *              https://www.sdcard.org/downloads/pls/pdf/part1_500.pdf.
+ *
+ * @author      Michel Rottleuthner <michel.rottleuthner@haw-hamburg.de>
+ */
+
+#ifndef SDCARD_SPI_INTERNAL_H
+#define SDCARD_SPI_INTERNAL_H
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#include "periph/spi.h"
+#include "periph/gpio.h"
+#include "stdbool.h"
+#include "sdcard_spi.h"
+
+
+/* number of clocks that should be applied to the card on init
+   before taking furter actions (see sd spec. 6.4.1.1 Power Up Time of Card) */
+#define SD_POWERSEQUENCE_CLOCK_COUNT 74
+
+#define SD_CARD_PREINIT_CLOCK_PERIOD_US 10 /* used to generate 100 kHz clock in init phase*/
+#define SD_CARD_WAIT_AFTER_POWER_UP_US  1000
+
+/* R1 response bits (see sd spec. 7.3.2.1 Format R1) */
+#define SD_R1_RESPONSE_PARAM_ERROR       0b01000000
```

Comment:
I hate to be nitty (: but binary literals are [GCC Extensions](https://gcc.gnu.org/onlinedocs/gcc/Binary-constants.html). I would vote for replacing them with hex numbers or with shifts+ors